substrate/client/authority-discovery/src/worker.rs

```diff
@@ -946,6 +982,44 @@ where
 	}
 }
 
+/// Removes the `/p2p/..` from the address if it is present.
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum AddressType {
+	/// The address is specified as a public address via the CLI.
+	PublicAddress(Multiaddr),
+	/// The address is a global listen address.
+	GlobalListenAddress(Multiaddr),
+	/// The address is discovered via the network (ie /identify protocol).
+	ExternalAddress(Multiaddr),
+}
+
+impl AddressType {
+	/// Removes the `/p2p/..` from the address if it is present.
+	///
+	/// In case the peer id in the address does not match the local peer id, an error is logged for
+	/// `ExternalAddress` and `GlobalListenAddress`.
+	fn without_p2p(self, local_peer_id: PeerId) -> Multiaddr {
+		// Get the address and the source str for logging.
+		let (mut address, source) = match self {
+			AddressType::PublicAddress(address) => (address, "public address"),
+			AddressType::GlobalListenAddress(address) => (address, "global listen address"),
+			AddressType::ExternalAddress(address) => (address, "external address"),
+		};
+
+		if let Some(multiaddr::Protocol::P2p(peer_id)) = address.iter().last() {
+			if peer_id != *local_peer_id.as_ref() {
+				error!(
```

Comment:
This line for sure breaks this test:
https://github.com/paritytech/polkadot-sdk/blob/52a7325e0e8f46a7914f7a868bd4b9b00f54f6c2/substrate/zombienet/0002-validators-warp-sync/test-validators-warp-sync.zndsl#L38

Does it need to be error? If so, I can adjust the condition in test  to be `1`.

Reply:
I can change it back to warning for now, sorry missed that 🙏

Reply:
The error is coming from the validator setup:
```
2024-11-05 20:07:15.454	
2024-11-05 18:07:15.454 ERROR tokio-runtime-worker sub-authority-discovery: No public addresses configured and no global listen addresses found. Authority DHT record may contain unreachable addresses. Consider setting `--public-addr` to the public IP address of this node. This will become a hard requirement in future versions for authorities.    
```

Will adjust the testing instead to accommodate this 🙏